src/main/java/org/mule/module/apikit/validation/body/schema/v2/RestSchemaV2Validator.java

```diff
@@ -27,9 +28,20 @@ public RestSchemaV2Validator(MimeType mimeType) {
   }
 
   public void validate(String payload) throws BadRequestException {
-    final List<ApiValidationResult> validationResults = mimeType.validate(payload);
-    if (!validationResults.isEmpty()) {
-      throw new BadRequestException(buildLogMessage(validationResults));
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    Future<List<ApiValidationResult>> future = executor.submit(() -> mimeType.validate(payload));
+
+    try {
+      final List<ApiValidationResult> validationResults = future.get(2, TimeUnit.SECONDS);
+      if (!validationResults.isEmpty()) {
```

Comment:
why does validate() take longer? Is it because AMFShapePayloadValidator.syncValidate() takes longer OR in creating this validator?

Reply:
Yes thats correct, AMFShapePayloadValidator.syncValidate() takes longer.